pkg/backup/backup.go

```diff
@@ -105,46 +105,22 @@ func resolveActions(mapper meta.RESTMapper, actions map[string]Action) (map[sche
 	return ret, nil
 }
 
-// resolveResources uses the RESTMapper to resolve resources to their fully-qualified group-resource
-// names. fn is invoked for each resolved resource. resolveResources returns a list of any resources that failed to resolve.
-func (ctx *backupContext) resolveResources(mapper meta.RESTMapper, resources []string, allowAll bool, fn func(string)) {
-	for _, resource := range resources {
-		if allowAll && resource == "*" {
-			fn("*")
-			return
-		}
-		gr, err := resolveGroupResource(mapper, resource)
-		if err != nil {
-			ctx.log("Unable to resolve resource %q: %v", resource, err)
-			continue
-		}
-		fn(gr.String())
-	}
-}
-
-// getResourceIncludesExcludes takes the lists of resources to include and exclude, uses the
-// RESTMapper to resolve them to fully-qualified group-resource names, and returns an
-// IncludesExcludes list.
-func (ctx *backupContext) getResourceIncludesExcludes(mapper meta.RESTMapper, includes, excludes []string) *collections.IncludesExcludes {
-	resources := collections.NewIncludesExcludes()
-
-	ctx.resolveResources(mapper, includes, true, func(s string) { resources.Includes(s) })
-	ctx.resolveResources(mapper, excludes, false, func(s string) { resources.Excludes(s) })
-
-	ctx.log("Including resources: %v", strings.Join(resources.GetIncludes(), ", "))
-	ctx.log("Excluding resources: %v", strings.Join(resources.GetExcludes(), ", "))
-
-	return resources
-}
+// getResourceIncludesExcludes takes the lists of resources to include and exclude from the
+// backup, uses the discovery helper to resolve them to fully-qualified group-resource names, and returns
+// an IncludesExcludes list.
+func (ctx *backupContext) getResourceIncludesExcludes(helper discovery.Helper, includes, excludes []string) *collections.IncludesExcludes {
```

Comment:
Would it be worth trying to move this into the collections package?

Reply:
Ultimately I want to do this resolution before anything gets passed to the backupper/restorer - so they just take an IncludesExcludes struct (though this would require something like an Options struct). OK if I tackle this as part of that refactoring, once we agree on what we want it to look like?

Reply:
Sure